deploy/charts/venafi-kubernetes-agent/values.yaml

```diff
@@ -192,6 +192,23 @@ config:
     name:
     key:
 
+  # -- Configure the Venafi Kubernetes Agent to load credentials using a
+  # VenafiConnection resource.
+  venafiConnection:
```

Comment:
Why is this under `config`? As far as I understand, the `config` value is passed as a YAML file using the flag `--agent-config-file` or `-c` for short:

```console
$ go run . agent --help
The agent will periodically gather data for the configured data
        gatherers and send it to a remote backend for evaluation

Usage:
  preflight agent [flags]
  preflight agent [command]

Available Commands:
  info        print several internal parameters of the agent
  rbac        print the agent's minimal RBAC manifest

Flags:
  -c, --agent-config-file agent.yaml     Config file location, default is agent.yaml in the current working directory. (default "./agent.yaml")
```

This means that the configuration file will now contain the field `venafiConnection`. I guess it doesn't matter because this field is skipped when the YAML file is parsed, but would it not make more sense to move this to outside of the `config`?

Another solution would be for me to implement the `venafiConnection` field in the config file so that the user can either use the flag `--venafi-connection=name`, or use the config file field `venafiConnection.name`. I could even drop the command line flag in that case.

WDYT?

Reply:
> Why is this under config?

Because that's how it's currently documented:
* https://gitlab.com/venafi/vaas/ua/clouddocs/-/merge_requests/1133

Reply:
Ah, yes.

I'll add this to the configuration API then, which means you will no longer have to pass `--venafi-connnection` and `--venafi-connnection-namespace` in the deployment. This should make much more sense this way.

Reply:
> This means that the configuration file that gets stored in the configmap agent-config will now contain the field venafiConnection

No. That's not how the config template works in this project. You're thinking of cert-manager I think:
 * https://github.com/jetstack/jetstack-secure/blob/master/deploy/charts/jetstack-agent/templates/configmap.yaml

Reply:
@maelvls I've moved the values around to hopefully make them more intuitive.
The venafiConnection.enabled value is now in the authentication section.
And the venafiConnection.include value is now in the crds section.

So the documentation would look like:

> Use the following Helm chart values to install the VenafiConnection CRDs and to enable VenafiConnection based authentication:

```yaml
config:
  clusterName: venafi-kubernetes-agent-e2e
  clusterDescription: |
      A kind cluster used for testing the venafi-kubernetes-agent.

authentication:
  venafiConnection:
    enabled: true

crds:
  venafiConnection:
    include: true
```

> Or use the following Helm install command:

```sh
helm install venafi-kubernetes-agent oci://registry.venafi.cloud/charts/venafi-kubernetes-agent \
  --create-namespace \
  --namespace "venafi" \
  --set config.clusterName="replace-with-your-cluster-name" \
  --set config.clusterDescription="replace-with-your-cluster-description" \
  --set authentication.venafiConnection.enabled=true \
  --set crds.venafiConnection.include=true
```

Reply:
This new structure is simple and clear, I like it.

I'm OK with the fact that `--set crds.venafiConnection.include=true` is a bit different in venafi-enhanced-issuer and approver-policy-enterprise (it would be `--set venafiConnection.include=true` on these two projects).